File: go/resistor-color/resistor_color.go
// Package resistorcolor provides utilities to work with resistor color codes.
// It includes functions to retrieve the list of valid resistor colors and
// to determine the resistance value associated with a specific color.
// This package is useful for applications involving electronics, circuit design,
// or educational tools related to resistors.
package resistorcolor

// INVALID_COLOR_CODE represents that the color code value was not found
const INVALID_COLOR_CODE = -1

var colors = []string{
	"black", "brown", "red", "orange", "yellow",
	"green", "blue", "violet", "grey", "white",
}

// Colors returns the list of all colors.
func Colors() []string {
	result := make([]string, len(colors))
	copy(result, colors)
	return result
}

// ColorCode returns the resistance value of the given color.
func ColorCode(color string) int {
	for i, c := range colors {
		if c == color {
			return i
		}
	}
	return INVALID_COLOR_CODE
}

File: php/resistor-color/ResistorColor.php
<?php

declare(strict_types=1);

/**
 * Returns a complete list of standard resistor color codes.
 *
 * The colors are ordered according to their numerical value in the resistor color code system,
 * starting from black (0) to white (9).
 *
 * @return array<string> An indexed array of strings, where each string is a resistor color name.
 */
function getAllColors(): array
{
    return [
        "black",
        "brown",
        "red",
        "orange",
        "yellow",
        "green",
        "blue",
        "violet",
        "grey",
        "white"
    ];
}

/**
 * Retrieves the numerical code for a given resistor color.
 *
 * This function looks up the provided color string in the standard list of resistor colors
 * and returns its corresponding integer value.
 *
 * @param string $color The name of the resistor color (e.g., "brown", "red").
 * @return int The numerical code associated with the color. Returns `false` if the color is not found,
 * though in a strictly typed context where all inputs are valid colors, this won't happen.
 *
 * @example
 * colorCode("black"); // Returns 0
 * colorCode("brown"); // Returns 1
 * colorCode("white"); // Returns 9
 */
function colorCode(string $color): int
{
    return array_search($color, getAllColors());
}